src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -135,6 +135,23 @@ public function create_item_permissions_check( $request ) {
 			);
 		}
 
+		// If the upload is an image, check if the server can handle the mime type.
+		$files = $request->get_file_params();
+		if (
+			! empty( $files ) &&
+			isset( $files['file'] ) &&
+			str_starts_with( $files['file']['type'], 'image/' )
+		) {
+			// Check if the image editor supports the type.
+			if ( ! wp_image_editor_supports( array( 'mime_type' => $files['file']['type'] ) ) ) {
+				return new WP_Error(
+					'rest_upload_image_type_not_supported',
+					__( 'The web server cannot generate responsive image sizes for this image. Convert it to JPEG or PNG before uploading.', 'gutenberg' ),
```

Comment:
Wrong text domain

Reply:
good catch, thanks!

Reply:
fixed in https://github.com/WordPress/wordpress-develop/pull/8322/commits/595a95d55f22205e92c0c5ccde24936b4740a5dc